### ISSUE TITLE
Defer 

### ISSUE THREAD

**Issue by tovare** [peterbe/govspy#5]
In Python it would be useful to list the

with open('workfile') as f:
     read_data = f.read()
 f.closed

In addition to or instead of the try except.
https://docs.python.org/3/tutorial/inputoutput.html

**Comment by peterbe** [comment]
I agree! I don't do anything like that with `open` in Python.

It would probably be better to do something entirely different with the Python example, in the `finally:` block.

**Comment by riggsd** [comment]
Agreed. The closest equivalent to Go's `defer` is not a `try/except`, but the context manager (aka `with` statement). See: https://www.python.org/dev/peps/pep-0343/

Of course, the closest thing to Python's `try/except` that we're supposed to use in Go _is_ the `defer` statement. You already touch on Go's `panic/recover`, which is closer in spirit to Python's `try/except`, though the `defer` statement is intended as more general purpose.
